bitcoin-magazine/bitcoin-explained/op_return.md

```diff
@@ -0,0 +1,596 @@
+But the difference is that this little database that you're keeping of the UTXO set, that no longer has to grow.
+And you can also normally throw away old blocks.
+So if you have a small hard drive, you can always toss out the old blocks, but you can never toss out anything from the UTXO set.
+
+Aaron Van Wirdum: 00:09:54
+
+Yeah, so the two main benefits are, first of all, storage, especially for pruned nodes that don't store the entire blockchain, because they now have a benefit because they have a smaller UTXO set.
+And the other one is computational.
+When a new transaction comes in, there's a smaller UTXO set to check the new transaction against.
+
+Sjors Provoost: 00:10:13
+
+Yeah.
+So I guess a better way to say that is it's saving you, potentially saving you memory.
+So ideally you want to keep the entire UTXO set in RAM.
+Now that's already not possible these days.
+I guess back when this discussion was happening, the UTXO set may have been much smaller, maybe a few hundred megabytes.
+Now it is over 10 gigabytes.
+So for now most people unless you have like you know 20 gigabytes of RAM you're probably not holding it in RAM anyway.
+But it is nice if you can keep the UTXO set in RAM all the time it's much faster.
+And we did an episode about that.
+It is episode 15 in which we explain a mechanism that could solve this problem called U3XO.
+And that would remove the need to keep the UTXO set in RAM, but it comes at the expense of some trade-offs.
+And we just explain those there.
+
+Aaron Van Wirdum: 00:11:02
+
+Okay, let's stick to the OP_RETURN for now.
+People were storing data in the blockchain.
+This came at a cost for people that were running full nodes.
+And so at some point, developers figured, okay, look guys, if you're going to store data in the blockchain anyways, then please use OP_RETURN.
+And that way, there's a smaller cost to all nodes.
+
+Sjors Provoost: 00:11:27
+
+Yeah, but there was a second request too, I would say, because this limit of 40 bytes.
+
+Aaron Van Wirdum: 00:11:32
+
+Yeah, I was going to get there.
+So OP_RETURN was always in the protocol, but it was basically useless because nodes wouldn't forward OP_RETURN transactions.
+Then Bitcoin developers said, all right guys, you're going to store data anyways.
+So we'll allow, we'll effectively enable OP_RETURN.
+But then the next debate was about the size of the OP_RETURN messages.
+
+## The OP_RETURN message size debate
+
+Sjors Provoost: 00:11:55
+
+Yeah, exactly.
+So one number you could have used for that size was to say, okay, how big can these multi-sig transactions be?
+And I think that was 172 bytes so you could have said okay we're just gonna make OP_RETURN 172 bytes because in that case there is really no reason to use these multi-sig keys anymore because you might as well use OP_RETURN but there was a simultaneous I guess movement or desire to not make the blockchain unnecessarily big.
+
+Aaron Van Wirdum: 00:12:21
+
+And
+
```

Comment:
This is another instance of misattribution to a speaker. Here, it's Sjors who says, 'And' In this case, it might be best to remove this particular line. Overall, it's helpful to eliminate filler words and unnecessary lines for a more concise transcript.